ntoskrnl/mm/section.c

```diff
@@ -5014,10 +5009,11 @@ MmFlushSegment(
     MmUnlockSectionSegment(Segment);
     MmDereferenceSegment(Segment);
 
+Quit:
     if (Iosb)
-        Iosb->Status = STATUS_SUCCESS;
+        Iosb->Status = Status;
 
-    return STATUS_SUCCESS;
+    return Status;
```

Comment:
These 2 changes and line 4957 addition are superfluous.

Reply:
You are indeed correct, but I think that @TAN-Gaming wanted to keep the provision where supplemental code could be added in-between (above the `Quit:`) in the future, which, in case of failure, could set `Status` to a different value.

Reply:
Currently, the code doesn't handle failure of the flush at all beside the length check. So, I used the `Status` variable here for the ease of improving in the future.

Reply:
I do agree with Serge here. Wait with that until it would become necessary. Currently it is not necessary, and reduces locality.

Reply:
> I don't understand that idea: the 2 replacements by goto are correct

The code paths are the same, why have multiple returns ?

Reply:
Updated the comment to describe the logic better.

> Undo'ing new 4957,4958, 5014,5016 should work as well, and is more local.

But, you will have to write the `if (Iosb) Iosb->Status = STATUS_SUCCESS;` check everywhere before `goto Quit` for the "Nothing to flush" code path. The `Status` variable already exists, why not just use it and set the value in one go ?

Reply:
> write the `if (Iosb) Iosb->Status = STATUS_SUCCESS;` check everywhere before `goto Quit;` for the "Nothing to flush" code path.

Why? The existing `Quit:` code already does that!

Reply:
> The existing Quit: code already does that!

I agree that hardcoding the status value there is OK for the current code that assume always success. But in term of logic, setting the initial status value at 4958 is better because it corresponds with the initial number of flushed bytes (set at 4960).

Reply:
> I agree that hardcoding the status value there is OK for the current code that assume always success.

That's all I noted.

> But in term of logic, setting initial status value at 4958 is better because it corresponds with the initialization of the number of flushed bytes (set at 4960).

I would argue with that:
I see little "logic" here, as `Status` value never changes (yet),
`Iosb->Information` initialization is a somewhat different case, as its value can be incremented.

Reply:
OK. Let's revert this part and move on.